client/state/sites/actions.js

```diff
@@ -107,9 +114,10 @@ export function requestSites() {
  * a site.
  *
  * @param  {Number}   siteId Site ID
+ * @param  {Boolean}  removeSiteOnFailure Remove the site from list if request fails
```

Comment:
Nitpicking: Let's keep the separate JSDoc sections aligned together vertically, in this case - `Site ID` with `Remove the site ...`.